docs/source-2.0/aws/aws-endpoints.rst

```diff
@@ -0,0 +1,375 @@
+    }
+
+
+
+.. smithy-trait:: aws.endpoints#standardRegionalEndpoints
+.. _aws.endpoints#standardRegionalEndpoints-trait:
+
+-------------------------------------------------
+``aws.endpoints#standardRegionalEndpoints`` trait
+-------------------------------------------------
+
+Summary
+    Indicates that a services endpoints should be resolved using the standard regional
+    patterns:
+
+    - Default: ``{service}.{region}.{dnsSuffix}``
+    - Fips: ``{service}-fips.{region}.{dnsSuffix}``
+    - Dualstack: ``{service}.{region}.{dualStackDnsSuffix}``
+    - Fips/Dualstack: ``{service}-fips.{region}.{dualStackDnsSuffix}``
+
+Trait selector
+    ``service``
+Trait value
+    A ``structure`` with the following properties:
+
+.. list-table::
+    :header-rows: 1
+    :widths: 10 30 60
+
+    * - Property
+      - Type
+      - Description
+    * - partitionSpecialCases
+      - ``List<PartitionSpecialCase>`` of `PartitionSpecialCase object`_
+      - A list of partition special cases - endpoints for a partition that do not follow the
+        standard patterns.
+    * - regionSpecialCases
+      - ``List<RegionSpecialCase>`` of `RegionSpecialCase object`_
+      - A list of regional special cases - endpoints for a region that do not follow the
+        standard patterns.
+
+Conflicts with
+    :ref:`aws.endpoints#nonRegionalizedEndpoints-trait`
+
+Most AWS services are regionalized and are strongly encouraged to follow
+the standard endpoint patterns defined above, both for consistency and to
+ensure that endpoints are forwards compatible and that SDK updates are
+not required when the service launches in a new region or partition.
+
+The following example defines a service that uses the standard regional endpoints:
+
+.. code-block:: smithy
+
+    $version: "2"
+
+    namespace smithy.example
+
+    use aws.endpoints#standardRegionalEndpoints
+
+    @standardRegionalEndpoints
+    service MyService {
+        version: "2020-04-02"
+    }
+
+While services are strongly encouraged to follow standard endpoint patterns,
+there are occasional exceptions and special cases.  The following example defines
+a service that use standard regional endpoints, but uses a non-standard pattern for
+FIPS endpoints in US GovCloud:
+
+.. code-block:: smithy
+
+    @standardRegionalEndpoints{
+        partitionSpecialCases: [
+            {
+                partition: "aws-us-gov",
+                endpoint: "myservice.{region}.{dnsSuffix}",
+                fips: true
+            }
+        ]
+    }
+    service MyService {
+        version: "2020-04-02"
+    }
+
+---------------------------
+PartitionSpecialCase object
+---------------------------
+
+A PartitionSpecialCase object contains the following properties:
+
+.. list-table::
+    :header-rows: 1
+    :widths: 10 30 60
+
+    * - Property name
+      - Type
+      - Description
+    * - partition
+      - ``string``
+      - **Required**. The partition to special case (example: "aws").
```

Comment:
Would it be beneficial to have this as a map key instead of as a member of an object in a list?

Reply:
I had debated this a bit.  The issue is with variants (fips/dualstack) - you would then need to have the key map to a list of special cases, so the example above would become:
```
@standardRegionalEndpoints{
    partitionSpecialCases: {
      "aws-us-gov": [
          {
             endpoint: "myservice.{region}.{dnsSuffix}",
             fips: true
          }
       ]
   }
}
```

The same applies with the regionalSpecialCases.  We can have either 1) A list of special cases that include the region/partition or 2) a map with key of region/partition and values that are a list of special cases.  

I landed on 1 as most of our special cases are for a single variant (as in the example above).  But I don't have a strong preference between these two.  Thoughts?